### PR TITLE
Add disable anti replay filter.

### PR DESCRIPTION
**Pull request by boaks**
Address issue #653

Signed-off-by: Achim Kraus <achim.kraus@bosch-si.com>

### CODE REVIEW

**Comment by boaks** [comment]
I would like to merge this PR at the end of the week.

File: scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java [hunk 1]
```diff
@@ -1641,6 +1681,40 @@ public Builder setNoServerSessionId(boolean flag) {
...
+		/**
+		 * Use window filter.
+		 * 
+		 * Messages too old for the filter window will pass the filter.
+		 * 
+		 * @param enable {@code true} to enable filter. Default {@code false}.
+		 * @return this builder for command chaining.
+		 * @throws IllegalArgumentException if anti replay window filter is active.
+		 * @see http://tools.ietf.org/html/rfc6347#section-4.1
+		 */
```
Comment (sbernard31) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java]: Is it a normal use case to do that ? I feel this is more a kind of workaround?
If I am right, should we precise this is not really intended to be used ? (or something like this)

Bonus question, not directly linked to this PR, does the spec support windows size > 64 ? (min size seems to be 32) I didn't check the windows sliding algorithm to know if my question makes sense :grin:

File: scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java [hunk 2]
```diff
@@ -1748,6 +1822,12 @@ public DtlsConnectorConfig build() {
 			if (config.sniEnabled == null) {
 				config.sniEnabled = Boolean.FALSE;
 			}
+			if (config.useAntiReplayFilter == null) {
+				config.useAntiReplayFilter = true;
+			}
+			if (config.useWindowFilter == null) {
+				config.useWindowFilter = false;
+			}
```
Comment (sbernard31) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java]: A detail, but it seems that setter does not allow a config where  `config.useAntiReplayFilter=true` and `config.useWindowFilter= true`

But if I set only userWindowsFilter to true, then the default value of useAntiReplayFilter will be true too.

File: scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java (hunk 1 above)
Comment (boaks) [reply scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java]: I guess, yes. But I don't see the benefit. Only 1 issue with that. Others seems not to have that issue. 64 is a long and more will make it more complex :-).

File: scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java (hunk 2 above)
Comment (boaks) [reply scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java]: True, that's irritating.

DTLSConnector

```
this.useWindowFilter = config.useWindowFilter();
this.useFilter = config.useAntiReplayFilter() || useWindowFilter;

```

so it doesn't harm. But I will change the checks. I didn't wont "ambiguity"  in the configuration.

**Pull request merged**